src/libraries/System.Private.CoreLib/src/System/Resources/ManifestBasedResourceGroveler.cs

```diff
@@ -34,6 +36,7 @@ namespace System.Resources
     internal partial class ManifestBasedResourceGroveler : IResourceGroveler
     {
         private readonly ResourceManager.ResourceManagerMediator _mediator;
+        private static bool AllowReflectionForNonPrimitiveTypes { get; } = AppContext.TryGetSwitch("System.Resources.AllowReflectionForNonPrimitiveTypes", out bool allowReflection) ? allowReflection : true;
```

Comment:
Can we add a test for setting this switch and verify that the correct exception is thrown?